lib/realtime_signs.ex

```diff
@@ -67,6 +68,14 @@ defmodule RealtimeSigns do
     end
   end
 
+  defp consume_message_logs do
+    if Application.get_env(:realtime_signs, :message_log_zip_url) do
+      [RealtimeSigns.Scheduler]
```

Comment:
I'm kind of thinking the `Scheduler` should just be considered a new piece of the Realtime Signs infrastructure that always starts up. It might be useful for other things going forward. (I think some of the recurring `Engine` updates could be mediated a little more clearly through Quantum, for example.)

And then it would be the jobs themselves that are dynamically determined by the presence of an environment variable. So you'll always have the runner; it's just which jobs in which environment will it run?

Reply:
Hmm yeah I can see how the scheduler could be useful other things! I've set up an approach where the job is now added dynamically. The thing I had some trouble with is that the job wasn't being added if I kept the Scheduler as a child process of the Supervisor since the Scheduler process wasn't running when I was calling my helper function that configures the job.

Curious as to what your thoughts would be for an approach to configuring and starting the job! What I have right now works, but I'm not certain how crucial it is to have the Scheduler running as a child process under the main Supervisor. Are you able to see a workaround where we can have the Scheduler first added as a child process and then configure/add the job after that point?

Reply:
Hmm, it could be that dynamically adding jobs is not really how it's made to be used, in which case I think probably the way you had it before is better, which was simpler and more straightforward to follow than what's there now, IMO.

What I had been envisioning was having the `RealtimeSigns.Scheduler` be unconditionally started by the supervision tree, and having the conditional be in `runtime.exs` where jobs were defined:

```elixir
scheduler_jobs =
  if System.get_env("MESSAGE_LOG_CRON_SCHEDULE") do
    [
      {
        System.get_env("MESSAGE_LOG_CRON_SCHEDULE"), 
        {RealtimeSigns.MessageLogJob, :get_and_store_logs, []}
      }
    ]
  else
    []
  end

config :realtime_signs, RealtimeSigns.Scheduler, jobs: scheduler_jobs
```

Does something like that not work?

Reply:
Ohh didn't think of binding the definition of the jobs to a variable like this. It works! Just updated the PR.